app/views/users/show.html.erb

```diff
@@ -4,7 +4,8 @@
   </div>
 </div>
 
-<% if current_user == @user || !@user.private? || current_user.leaders.include?(@user) %>
+
+<% if policy(@user).show? %>
```

Comment:
That should probably use another policy, as `.show?` handles authorization to render this whole page, this conditional on this block of code does not get triggered